canard.c

```diff
@@ -84,24 +86,30 @@ void canardInit(CanardInstance* out_ins,
      * If your application fails here, make sure it's not built in 64-bit mode.
      * Refer to the design documentation for more info.
      */
-    CANARD_ASSERT(CANARD_MULTIFRAME_RX_PAYLOAD_HEAD_SIZE >= 6);
+    CANARD_ASSERT(CANARD_MULTIFRAME_RX_PAYLOAD_HEAD_SIZE(CANARD_MEM_BLOCK_SIZE_BASE) >= 2);
 
     memset(out_ins, 0, sizeof(*out_ins));
 
+    out_ins->transport_protocol = transport_protocol;
     out_ins->node_id = CANARD_BROADCAST_NODE_ID;
     out_ins->on_reception = on_reception;
     out_ins->should_accept = should_accept;
     out_ins->rx_states = NULL;
     out_ins->tx_queue = NULL;
     out_ins->user_reference = user_reference;
 
-    size_t pool_capacity = mem_arena_size / CANARD_MEM_BLOCK_SIZE;
-    if (pool_capacity > 0xFFFFU)
+    size_t block_size;
+
+    switch (transport_protocol)
```

Comment:
It seems like you need to propagate errors any place you have to branch on the transport_protocol. Otherwise I'm not sure how you handle the default cases here and elsewhere.

Reply:
On the default cases: Would it be terrible to say that passing anything else than what is defined as variants in the enum is undefined behavior?

Of course attempting to read an fd frame from a can2.0b would trigger an error. This error should ideally be a compile time error, but since this is C and little type safety exists, this is not straigh forward. The alternative would be to not trigger any error and call it UB.

Reply:
MISRA requires that every switch statement must have a default clause. If not sure what to put there, put `assert(false)` and/or return some error code.